crates/topos/src/components/node/mod.rs

```diff
@@ -79,13 +79,30 @@ pub(crate) async fn handle_command(
                 println!("Init the node without polygon-edge process...");
             } else {
                 // Generate the Edge configuration
-                if let Ok(result) = services::process::generate_edge_config(
+                match services::process::generate_edge_config(
                     edge_path.join(BINARY_NAME),
                     node_path.clone(),
                 )
                 .await
                 {
-                    if result.is_err() {
+                    Ok(Ok(status)) => {
+                        if let Some(0) = status.code() {
+                            println!("Edge configuration successfully generated");
+                        } else {
+                            println!(
+                                "Edge configuration generation terminated with error status: {:?}",
+                                status
+                            );
+                            remove_dir_all(node_path).expect("failed to remove config folder");
+                            std::process::exit(1);
+                        }
+                    }
+                    Ok(Err(e)) => {
+                        println!("Failed to generate edge config with error {e}");
+                        remove_dir_all(node_path).expect("failed to remove config folder");
+                        std::process::exit(1);
+                    }
+                    Err(e) => {
                         println!("Failed to generate edge config");
                         remove_dir_all(node_path).expect("failed to remove config folder");
                         std::process::exit(1);
```

Comment:
```suggestion
                    Ok(Err(e)) | Err(e) => {
                        println!("Failed to generate edge config with error {e}");
                        remove_dir_all(node_path).expect("failed to remove config folder");
                        std::process::exit(1);
```

I think you can simplify this with that

Reply:
Tried it already, compiler gives:
```
 this expression has type `std::result::Result<std::result::Result<std::process::ExitStatus, components::node::services::process::Errors>, tokio::task::JoinError>`
...
100 |                       Ok(Err(e)) |  Err(e) => {
    |                              -          ^ expected `Errors`, found `JoinError`
```

Reply:
Ah it is a `JoinError` ?
Maybe you can remove the `JoinHandle` and transform `generate_edge_config` into an `async` function.